### DISCUSSION TITLE
Error gradle for run cooja simulator

### DISCUSSION THREAD

**Discussion by zahrataghipour** [contiki-ng/contiki-ng#2844] [Q&A] [answered]
hi all
I have an error and I am posting the picture. Thank you for guiding me how to solve this problem.
![Screenshot (338)](https://github.com/contiki-ng/contiki-ng/assets/106019540/0f304c4e-ad48-4d28-a5a9-a036750420a4)
![Screenshot (338)](https://github.com/contiki-ng/contiki-ng/assets/106019540/b4bfee0c-db40-4165-aa56-afbf134fa59a)

**Comment by nfi** [comment]
It looks like it failed to fetch data from jcenter.bintray.com. I tried to fetch the file, which worked fine for me, so I guess there might have been a temporary issue with the server or your Internet connection. Do you still have the same issue?

Although still available as read-only, the jcenter service has been shut down, so I recommend upgrading to the latest version of Cooja that no longer uses the jcenter service.

**Reply by zahrataghipour** [reply]
Thank you for replying.
Yes, I still have the same issue and unfortunately it has not been solved. Excuse me, from which source should I download the latest version of the cooja simulator?
I want to work with cooja on contiki-ng

**Reply by nfi** [reply]
Just update to latest Contiki-NG. Cooja is included as a git submodule.

```
cd ~/contiki-ng
git pull
git submodule update
```

**Reply by zahrataghipour** [reply]
I also sent the picture of the commands , it is below

**Comment by zahrataghipour** [comment]
I executed the commands and now I got another error And I searched for the
error and realized that I should type this command in .bashrc , but it
still didn't solve it

**Reply by zahrataghipour** [reply]
![Uploading Screenshot (341).png…]()
![Uploading Screenshot (342)_LI.jpg…]()

**Reply by zahrataghipour** [reply]
![Screenshot (341)](https://github.com/contiki-ng/contiki-ng/assets/106019540/12d56544-f7f5-4bed-97c2-5a5307640581)
![Screenshot (342)_LI](https://github.com/contiki-ng/contiki-ng/assets/106019540/eb83b87a-a733-4779-ab35-4bedf783d61b)

**Reply by nfi** [reply, marked as answer]
It looks like `JAVA_HOME` is incorrectly set. It should not refer to a binary file but to the installation directory for Java. I suggest you test to remove the `JAVA_HOME` environment variable and let Gradle handle the configuration of a compatible JVM.

```
unset JAVA_HOME
./gradlew run
```
